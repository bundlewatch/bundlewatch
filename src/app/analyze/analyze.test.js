import { getOverallDifference } from '.'
import { mockFileResults } from './analyze.test.mockdata'

describe('getOverallDifference', () => {
    describe('Results match snapshot for the same input', () => {
        Object.keys(mockFileResults).forEach(mockFileResult => {
            it(mockFileResult, () => {
                expect(
                    getOverallDifference(
                        mockFileResults.oneFileLargerOneFileSmallerOverallSmaller,
                    ),
                ).toMatchSnapshot()
            })
        })
    })

    it('Returns bytes added when the file is larger than the base one', () => {
        const overallDifference = getOverallDifference(
            mockFileResults.oneFileLarger,
        )
        expect(overallDifference.totalAdded).toEqual(25)
        expect(overallDifference.totalRemoved).toEqual(0)
    })

    it('Returns bytes removed when the file is smaller than the base one', () => {
        const overallDifference = getOverallDifference(
            mockFileResults.oneFileSmaller,
        )
        expect(overallDifference.totalAdded).toEqual(150)
        expect(overallDifference.totalRemoved).toEqual(0)
    })

    it('Returns both bytes added and removed when some files are larger and some are smaller', () => {
        const overallDifference = getOverallDifference(
            mockFileResults.oneFileLargerOneFileSmallerOverallSmaller,
        )
        expect(overallDifference.totalAdded).toEqual(25)
        expect(overallDifference.totalRemoved).toEqual(50)
    })

    it('Returns a positive percentage change when the overall bundle size is larger', () => {
        const overallDifference = getOverallDifference(
            mockFileResults.oneFileLargerOneFileSmallerOverallLarger,
        )
        expect(overallDifference.percentageChange).toEqual(9.375)
    })

    it('Returns a negative percentage change when the overall bundle size is smaller', () => {
        const overallDifference = getOverallDifference(
            mockFileResults.oneFileLargerOneFileSmallerOverallSmaller,
        )
        expect(overallDifference.percentageChange).toEqual(-3.125)
    })
})
