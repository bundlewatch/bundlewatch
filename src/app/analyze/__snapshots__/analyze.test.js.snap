// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOverallDifference Results match snapshot for the same input oneFileLarger 1`] = `
Object {
  "percentageChange": -3.125,
  "totalAdded": 25,
  "totalRemoved": 50,
}
`;

exports[`getOverallDifference Results match snapshot for the same input oneFileLargerOneFileSmallerOverallLarger 1`] = `
Object {
  "percentageChange": -3.125,
  "totalAdded": 25,
  "totalRemoved": 50,
}
`;

exports[`getOverallDifference Results match snapshot for the same input oneFileLargerOneFileSmallerOverallSmaller 1`] = `
Object {
  "percentageChange": -3.125,
  "totalAdded": 25,
  "totalRemoved": 50,
}
`;

exports[`getOverallDifference Results match snapshot for the same input oneFileSmaller 1`] = `
Object {
  "percentageChange": -3.125,
  "totalAdded": 25,
  "totalRemoved": 50,
}
`;
